sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/TelemetryItemTests.cs

```diff
@@ -427,100 +433,92 @@ public void OTelResourceMetricTelemetryHasAllResourceAttributes()
                     { "foo", "bar" }
                 };
 
-                using ActivitySource activitySource = new ActivitySource(ActivitySourceName);
-                using var activity = activitySource.StartActivity(
-                    ActivityName,
-                    ActivityKind.Server,
-                    null,
-                    startTime: DateTime.UtcNow);
+            using ActivitySource activitySource = new ActivitySource(ActivitySourceName);
+            using var activity = activitySource.StartActivity(
+                ActivityName,
+                ActivityKind.Server,
+                null,
+                startTime: DateTime.UtcNow);
 
-                Assert.NotNull(activity);
+            Assert.NotNull(activity);
 
-                var resource = ResourceBuilder.CreateEmpty().AddAttributes(testAttributes).Build();
-                var azMonResource = resource.CreateAzureMonitorResource(instrumentationKey);
+            var resource = ResourceBuilder.CreateEmpty().AddAttributes(testAttributes).Build();
+            var azMonResource = resource.CreateAzureMonitorResource(platform: GetMockPlatform("true"), instrumentationKey: instrumentationKey);
 
-                var telemetryItems = TraceHelper.OtelToAzureMonitorTrace(new Batch<Activity>(new Activity[] { activity }, 1), null, instrumentationKey, 1.0f);
-                var telemetryItem = telemetryItems.FirstOrDefault();
+            var telemetryItems = TraceHelper.OtelToAzureMonitorTrace(new Batch<Activity>(new Activity[] { activity }, 1), azMonResource, instrumentationKey, 1.0f);
+            var telemetryItem = telemetryItems.First(x => x.Name == "Metric"); // Collect the "_OTELRESOURCE_" metric.
```

Comment:
This is a fix for a test bug. Ping me offline, we can debug and see this test fail in the `main` branch.